Resolve some build failures in Cookstyle

tas50 opened the pull request:
A few technically breaking changes, but this gets things in a better shape.

tas50 committed:
Disable the definitions Cookstyle cops

Fixing this requires a complete rewrite

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Require a working Apache cookbook release

7.0 breaks this cookbook

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Remove some checks for legacy RHEL/Debian/Ubuntu releases

None of this is necessary at this point and the rhel check was wrong

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Resolve a cookstyle warning by removing a legacy pagerduty fix

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
No need for package to be an array here

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Use multipackage installs to speed up installs

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Add changelog

Signed-off-by: Tim Smith <tsmith@chef.io>

tas50 committed:
Resolve yaml warnings

Signed-off-by: Tim Smith <tsmith@chef.io>

ramereth committed:
Disable spec tests due to definitions being used

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Use modern platforms for testing

Also remove deprecated require_chef_omnibus option.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Pin to apache2 < 6.0 as 7.0 also removed attributes

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Create helpers library to better manage platform configuration

This is an initial first step towards preparing this cookbook to be a resource
driven cookbook by moving various platform specific settings from attributes to
library helpers. While this is still using attributes, the core logic is now
done in in the helpers library.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Remove support for Debian 9 and Amazon Linux 2

Debian 9 does not provide a binary package for nagios nor does Amazon Linux 2.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix strange nagios issues with attributes in test-kitchen

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Remove distro configuration file on Debian 10

This is duplicating the configuration we set and causing issues.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Debian 10 and Ubuntu 20.04 use nagios4 to use vname

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix InSpec tests

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Cookstyle fixes

Signed-off-by: Lance Albertson <lance@osuosl.org>

kitchen-porter commented:
<!--
  0 Errors
  1 Warning: This is a big Pull Request.
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This is a big Pull Request.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

ramereth committed:
Update source build to nagios-4.4.6

- Make some fixes for the various platforms
- Re-enable InSpec tests.
- Default to using "nagios" as the vname for all directories, etc
- Use creates instead of not_if with File.exist?
- Switch to using execute for compile-nagios

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Ensure we install the cgis when building from source

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix tests for swappable-config suite

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Enable and fix tests for pagerduty

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Remove allowed-ips suite as that should be tested with ChefSpec

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Switch to using php cookbook and fix nginx cookbook version

In addition, remove use of apache attributes since those will go away
eventually.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix test for nginx on centos

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Additional helper methods for platform packages and paths

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Set sensitive for debconf-set-selections execute resource

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Remove support for Apache 2.2

Since we are now requiring apache2 ~> 5.0, we no longer support Apache 2.2 and
thus we can remove the use of the node['apache']['version'] attribute which no
longer exists.

This resolves #556.

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Update README

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
yamlint fixes

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth committed:
Fix delivery issues

Signed-off-by: Lance Albertson <lance@osuosl.org>

ramereth commented:
@tas50 can you please a look at the changes I included? I think this is ready to go!

ramereth committed:
cookstyle fixes

Signed-off-by: Lance Albertson <lance@osuosl.org>
